core/src/main/java/io/opentelemetry/android/instrumentation/InstallationContext.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import android.app.Application
+import io.opentelemetry.android.internal.services.ServiceManager
+import io.opentelemetry.api.OpenTelemetry
+
+data class InstallationContext(
+    val application: Application,
+    val openTelemetry: OpenTelemetry,
+    val serviceManager: ServiceManager,
```

Comment:
Can we make the application and serviceManager nullable as they're not needed in all instrumentations?

Reply:
Possibly, but I think we mostly want to use the same context when installing all instrumentations at startup -- in which case _some_ of the instrumentations will need some of these, so they won't be null. 

I think it's simpler if instrumentations just ignore (don't use) the parts they don't need.

Reply:
That makes sense! We can perhaps provide a getter for a no-op ServiceManager instance to support the use cases when a user is utilizing one/some of the many instrumentations like the okhttp3 instrumentation that does not require a ServiceManager?

Reply:
Based on yesterday's SIG meeting, we should try to find a way to allow instrumentations to reuse services/utilities without having to expose them as public apis. There's more details about it all on [this issue](https://github.com/open-telemetry/opentelemetry-android/issues/669).